mobile/native/src/channel_fee.rs

```diff
@@ -23,142 +25,141 @@ const WAIT_FOR_OUTBOUND_CAPACITY_TIMEOUT: Duration = Duration::from_secs(60);
-            );
-            Handle::current().block_on(async {
-                self.wait_for_outbound_capacity(channel_id, funding_tx_fees_msats)
-                    .await?;
-                // We add another sleep to ensure that the channel has actually been updated after
-                // receiving the payment. Note, this is by no means ideal and should
-                // be revisited some other time.
-                tokio::time::sleep(Duration::from_millis(500)).await;
-                anyhow::Ok(())
-            })
-        })
-        .context("Failed during wait for outbound capacity")?;
-
-        tracing::debug!(
-            "Trying to pay channel opening fees of {} sats",
-            transaction.fee
-        );
-        let funding_txid = transaction.txid;
-
-        if funding_tx_fees_msats > amount_msats {
-            tracing::warn!("Trying to pay fees with an amount smaller than the fees!")
-        }
-
-        let invoice_str = tokio::task::block_in_place(|| {
-            Handle::current().block_on(fetch_funding_transaction_fee_invoice(
-                transaction.fee,
-                funding_txid,
-            ))
-        })?;
-
-        match ln_dlc::send_payment(&invoice_str) {
-            Ok(_) => {
-                // unset the funding transaction marking it as being paid.
-                self.unset_open_channel_info();
-                tracing::info!("Successfully triggered funding transaction fees payment of {funding_tx_fees_msats} msats to {}", config::get_coordinator_info().pubkey);
-            }
-            Err(e) => {
-                tracing::error!("Failed to pay funding transaction fees of {funding_tx_fees_msats} msats to {}. Error: {e:#}", config::get_coordinator_info().pubkey);
-            }
-        };
+    /// Attempts to pay the fees for opening an inbound channel.
+    fn pay_jit_channel_open_fee(&self, amount_msats: u64) -> Result<()> {
+        let channels = self.channel_manager.list_channels();
+        // Assuming the user ever only has one channel. Needs to be changed when we are supporting
+        // multiple open channels at the same time.
+        let channel_details = channels.first();
+        match channel_details {
+            Some(channel_details) => {
+                let user_channel_id = UserChannelId::from(channel_details.user_channel_id);
+                let mut channel =
+                    db::get_channel(&user_channel_id.to_string())?.with_context(|| {
+                        format!("Couldn't find channel by user_channel_id {user_channel_id}")
+                    })?;
+
+                if channel.channel_state != ChannelState::OpenUnpaid {
+                    tracing::debug!("Channel inbound fees have already been paid. Skipping.");
+                    return Ok(());
+                }
 
-        Ok(())
-    }
+                let liquidity_options = tokio::task::block_in_place(ln_dlc::liquidity_options)?;
```

Comment:
Why does this have to be inside a `tokio::task::block_in_place`?

Reply:
what @luckysori said 🙂